documentation/design-docs/data-catalog.md

```diff
@@ -0,0 +1,497 @@
+
+<details>
+<summary>Click here for raw notes on dbt</summary>
+<br>
+
+* Requirements
+  * Referential integrity
+    * Built in via [tests](https://docs.getdbt.com/docs/build/tests) (relationship)
+  * Schema consistency
+    * [Ref versioning](https://docs.getdbt.com/reference/dbt-jinja-functions/ref#versioned-ref)
+    * [Snapshots](https://docs.getdbt.com/docs/build/snapshots)
+  * Documentation
+    * [Docs](https://docs.getdbt.com/docs/collaborate/documentation)
+    * Stored separately from the code, in YAML file
+      * Although [SQL comments](https://docs.getdbt.com/sql-reference/comments) are supported, and inline descriptions of databases/tables/views are allowed
+    * `dbt docs generate` compiles docs, `dbt docs serve` runs a server
+      * [Sample docs](https://www.getdbt.com/mrr-playbook)
+    * There are also [contracts](https://docs.getdbt.com/docs/collaborate/govern/model-contracts#how-are-contracts-different-from-tests)
+      * But these seem too strict for our use case (basically build failures)
+        * We would likely want to set `enforced: false`
+  * Orchestration
+    * Nothing built in, some recommended solutions include:
+      * Airflow
+      * [Prefect](https://www.prefect.io/)
+      * [Dagster](https://dagster.io/)
+      * [Meltano](https://docs.meltano.com/getting-started/meltano-at-a-glance)
+  * Views and joins
+    * [Docs](https://docs.getdbt.com/terms/view)
+    * Can be defined and documented like other tables
+  * Data validation
+    * [Tests](https://docs.getdbt.com/docs/build/tests) can be any SQL query
+    * Two failiure levels: `fail` and `warn`
+  * Automated flagging
+    * Nothing built in; we would have to produce output views and then use the
+      orchestration layer to define an automated process
+  * Monitoring
+    * Nothing built in, this will likely need to be part of the orchestration layer
+* We would have to use a [community adapter](https://dbt-athena.github.io/) to connect to Athena
+  * Migration may be challenging
+  * Not a [verified adapter](https://docs.getdbt.com/docs/supported-data-platforms#verified-adapters)
+* Web UI seems [expensive](https://www.getdbt.com/pricing/)
+  * Team is $100/seat/mo, and we likely need Enterprise features
+    * E.g. Team plan only allows 5 readonly users
+  * Core example uses [Meltano](https://docs.meltano.com/getting-started/meltano-at-a-glance) for ETL
+* A lot of useful stuff is only available in Cloud, including:
+  * CI/CD
+  * Job scheduling
+  * Deploy environments
+  * Built-in IDE
+  * Job monitoring
+* There are clearly [a lot of orchestrators](https://www.reddit.com/r/dataengineering/comments/10x4n7n/best_orchestration_tool_to_run_dbt_projects/)!
+
+</details>
+
+## Alternatives considered
+
+### [AWS Glue](https://aws.amazon.com/glue/)
+
+AWS Glue is an AWS service for ETL. It also provides utilities for data
+quality checks and data cataloging.
+
+#### Pros
+* Built-in to AWS, which we are already locked into
```

Comment:
Not strictly true. We _could_ switch to local Trino or DuckDB if our needs demanded it.